drivers/docker/config.go

```diff
@@ -652,6 +652,11 @@ func (d *Driver) SetConfig(c *base.Config) error {
 	}
 
 	d.config = &config
+	// Remove any http
+	if strings.HasPrefix(d.config.InfraImage, "https://") {
+		d.config.InfraImage = strings.Replace(d.config.InfraImage, "https://", "", 1)
+	}
```

Comment:
We perform this operation on the main image just before starting the task. It feels like it'd make sense to either:
* Do this in `driver/docker/network.go` where we pull the infra image.
* Do _both_ the main image and infra image in this spot.

Reply:
https://github.com/hashicorp/nomad/blob/a40a14064a045d7ed6d2dc675ef8c45998f70cf0/drivers/docker/network.go#L22
I actually tried to put it in `network.go` at beginning haha. But quite some methods read `*d.config.InfraImage directly`, so if I want those methods to take the change, I need to mutate global struct state. If I mutate the global state, if doesn't feel right to do it every time before pull infra. So I put it here.

@tgross could you help to put it where you think fits most? It's a minor inconsistency, which was causing some confusion to us, but def not a blocker.

Reply:
> If I mutate the global state, if doesn't feel right to do it every time before pull infra. 

Yeah, I'm inclined to agree with you there. I know we had a couple PRs recently around the infrastructure images, so I want to make sure I get the opinion of folks who worked on that. @shoenig what do you think about moving the `http://` stripping for the infra images here too?

Reply:
Seems fine to me. I do feel that both code blocks should be using https://golang.org/pkg/strings/#TrimPrefix

Reply:
Good call. @fredwangwang sorry about the delay in getting back on this but do you think you can make that change? If not, happy to carry this PR.

Reply:
go for it @tgross, and thanks for getting back to this issue!

Reply:
It turns out that those two sections of code didn't both have access to the same variables, so it doesn't make sense to handle them in the same place after all. I've made @shoenig's suggestion to use `TrimPrefix`, rebased this PR on master, and added a changelog entry.